Remotion/Data/DomainObjects.UnitTests/Mapping/Validation/TypeDefinitionValidatorTest.cs

```diff
@@ -0,0 +1,136 @@
+
+      _validationRuleMock1.Setup(mock => mock.Validate(_typeDefinition1)).Returns(_fakeValidMappingValidationResult).Verifiable();
+
+      var mappingValidationResults = validator.Validate(new[] { _typeDefinition1 }).ToArray();
+
+      _validationRuleMock1.Verify();
+      Assert.That(validator.ValidationRules.Count, Is.EqualTo(1));
+      Assert.That(mappingValidationResults.Length, Is.EqualTo(0));
+    }
+
+    [Test]
+    public void ValidateWithOneRuleAndClassDefinition_InvalidResult ()
+    {
+      var validator = new TypeDefinitionValidator(_validationRuleMock1.Object);
+
+      _validationRuleMock1.Setup(mock => mock.Validate(_typeDefinition1)).Returns(_fakeInvalidMappingValidationResult).Verifiable();
+
+      var mappingValidationResults = validator.Validate(new[] { _typeDefinition1 }).ToArray();
+
+      _validationRuleMock1.Verify();
+      Assert.That(validator.ValidationRules.Count, Is.EqualTo(1));
+      Assert.That(mappingValidationResults.Length, Is.EqualTo(1));
+      Assert.That(mappingValidationResults[0], Is.SameAs(_fakeInvalidMappingValidationResult));
+    }
+
+    [Test]
+    public void ValidateWithSeveralRulesAndClassDefinitions_ValidResult ()
+    {
+      var validator = new TypeDefinitionValidator(_validationRuleMock1.Object, _validationRuleMock2.Object, _validationRuleMock3.Object);
+
+      _validationRuleMock1.Setup(mock => mock.Validate(_typeDefinition1)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock1.Setup(mock => mock.Validate(_typeDefinition2)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock1.Setup(mock => mock.Validate(_typeDefinition3)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock2.Setup(mock => mock.Validate(_typeDefinition1)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock2.Setup(mock => mock.Validate(_typeDefinition2)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock2.Setup(mock => mock.Validate(_typeDefinition3)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock3.Setup(mock => mock.Validate(_typeDefinition1)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock3.Setup(mock => mock.Validate(_typeDefinition2)).Returns(_fakeValidMappingValidationResult).Verifiable();
+      _validationRuleMock3.Setup(mock => mock.Validate(_typeDefinition3)).Returns(_fakeValidMappingValidationResult).Verifiable();
+
+      var mappingValidationResults = validator.Validate(new[] { _typeDefinition1, _typeDefinition2, _typeDefinition3 }).ToArray();
+
+      _validationRuleMock1.Verify();
+      Assert.That(validator.ValidationRules.Count, Is.EqualTo(3));
+      Assert.That(mappingValidationResults.Length, Is.EqualTo(0));
+    }
+
+    [Test]
+    public void ValidateWithSeveralRulesAndClassDefinitions_InvalidResult ()
```

Comment:
Those two tests don't say 'why' they're once invalid and once value. Is this the same with other validator tests or just here? if it's just here, please fix. otherwise, please point me towards them :)

Reply:
I just renamed the file no idea what the test were supposed to do :p

Looking at the other validator tests, for example `SortExpressionValidatorTest`, they have the same structure.